rotkehlchen/data_import/importers/uphold_transactions.py

```diff
@@ -115,19 +116,24 @@ def _consume_uphold_transaction(
                     raise SkippedCSVEntry(f'Trade destination amount is {destination_amount}.')
         elif origin == 'uphold' and transaction_type == 'out':
             if origin_asset == destination_asset:  # Withdrawals
-                asset_movement = AssetMovement(
+                events = [AssetMovement(
                     location=Location.UPHOLD,
-                    category=AssetMovementCategory.WITHDRAWAL,
-                    address=None,
-                    transaction_id=None,
-                    timestamp=timestamp,
+                    event_type=HistoryEventType.WITHDRAWAL,
+                    timestamp=ts_sec_to_ms(timestamp),
                     asset=origin_asset,
-                    amount=origin_amount,
-                    fee=Fee(fee),
-                    fee_asset=fee_asset,
-                    link='',
-                )
-                self.add_asset_movement(write_cursor, asset_movement)
+                    balance=Balance(origin_amount),
+                )]
+                if fee != ZERO:
+                    events.append(AssetMovement(
+                        event_identifier=events[0].event_identifier,
+                        location=Location.UPHOLD,
+                        event_type=HistoryEventType.WITHDRAWAL,
+                        timestamp=ts_sec_to_ms(timestamp),
+                        asset=fee_asset,
+                        balance=Balance(fee),
+                        is_fee=True,
```

Comment:
so question. In those I see neither unique id, nor the thing you called memo before. So how would you identify duplicates if I say .... run the import of the uphold CSV twice? There is a fundamental issue with all of those that you would need to address

Reply:
As mentioned in my response to the other comment about duplicates: https://github.com/rotki/rotki/pull/9032#discussion_r1873511852, using a unique constraint should handle this correctly. If these events are imported a second time they will fail the unique contstraint, as they will be identical to what is already in the database.